tests/functional/test_install_reqs.py

```diff
@@ -10,6 +10,12 @@
 from tests.lib.local_repos import local_checkout
 
 
+if os.name == 'posix':
+    distutils_cfg = os.path.expanduser('~/.pydistutils.cfg')
+else:
+    distutils_cfg = os.path.expanduser('~\\.pydistutils.cfg')
+
+
```

Comment:
`/` works on both POSIX and Windows, so no need for this conditional. On the other hand, you've changed the Windows filename to now include an initial . - was that deliberate or is this a bug? If it is a bug, the tests didn't pick it up so can you look at the tests to confirm why?

Reply:
Changing was probably a copy-paste error. Why it doesn't fail is another question.

From https://github.com/python/cpython/blob/6f0eb93183519024cb360162bdd81b9faec97ba6/Lib/distutils/dist.py#L348 configs on Linux and Windows files are different, so 4 tests together that use this setting including https://github.com/techtonik/pip/blob/b395355b0188e64d0224a744b632333a5e3144d7/tests/functional/test_install_reqs.py#L232 are probably indifferent to those settings.

Reply:
I can't look into this, because I ditched Windows from my life.

Reply:
On the second thought at least it should be possible to make them fail on both platforms by renaming the file to some nonsense.

Reply:
You should be able to write a test that fails on Appveyor, then modify the code to resolve the test failure. Agreed, it's more painful if you don't have a local Windows system, but it's still doable.

On the other hand, this all seems unrelated to the original point of the PR, which is simply about making the options mutually exclusive. Can't the original requirement be satisfied without changing this area of code at all? Sorry if I missed something, but there's no obvious comment in the thread explaining why we need to modify this.

Reply:
I can't run the single test due to #4878. `tox` doesn't allow this too, right?